### PR TITLE
expression: implement vectorized evaluation for `JSONArrayAppend` 

### LINKED ISSUE #12104: Call For Participation: vectorize all JSON builtin functions
This issue is used to trace all vectorized JSON builtin functions, which is a part of https://github.com/pingcap/tidb/issues/12058.

**NOTE**:
1. Feel free to join us in the [`wg-vec-expr` working group](https://github.com/pingcap/community/blob/master/working-groups/wg-vec-expr.md)
2. Please comment on this issue if you are working on any of the functions.

**Note For Performance Challenge Teams: each PR on this issue has 50 score.**

-----
- [x] builtinCastIntAsJSONSig
- [x] builtinCastRealAsJSONSig
- [x] builtinCastDecimalAsJSONSig
- [x] builtinCastStringAsJSONSig
- [x] builtinCastDurationAsJSONSig
- [x] builtinCastTimeAsJSONSig
- [x] builtinCastJSONAsJSONSig
- [x] builtinCoalesceJSONSig
- [x] builtinCaseWhenJSONSig
- [x] builtinIfJSONSig @Reminiscent https://github.com/pingcap/tidb/pull/12081
- [x] builtinIfNullJSONSig
- [x] builtinJSONExtractSig
- [x] builtinJSONSetSig
- [x] builtinJSONInsertSig
- [x] builtinJSONReplaceSig
- [x] builtinJSONRemoveSig
- [x] builtinJSONMergeSig
- [x] builtinJSONObjectSig
- [x] builtinJSONArraySig
- [ ] builtinJSONArrayAppendSig
- [x] builtinJSONArrayInsertSig
- [ ] builtinJSONSearchSig
- [x] builtinJSONKeysSig
- [x] builtinJSONKeys2ArgsSig
- [x] builtinJSONAnyValueSig
- [x] builtinNameConstJSONSig
- [x] builtinJSONUnquoteSig

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #12104 

Problem Summary: implement vectorized evaluation for `JSONArrayAppend`

### What is changed and how it works?
implement vectorized evaluation for `JSONArrayAppend`

```
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONArrayAppendSig-VecBuiltinFunc-12               868           1280255 ns/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONArrayAppendSig-NonVecBuiltinFunc-12            920           1284265 ns/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONArrayAppendSig-VecBuiltinFunc#01-12            481           2462974 ns/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONArrayAppendSig-NonVecBuiltinFunc#01-12         470           2485364 ns/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONArrayAppendSig-VecBuiltinFunc#02-12            692           1786525 ns/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONArrayAppendSig-NonVecBuiltinFunc#02-12         680           1751424 ns/op
```

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: expression/builtin_json_vec.go [hunk 1]
```diff
@@ -967,11 +967,83 @@ func (b *builtinJSONContainsPathSig) vecEvalInt(input *chunk.Chunk, result *chun
...
+		for j := 0; j < m; j++ {
+			s, sNull := "", pathBufs[j].IsNull(i)
+			if !sNull {
+				s = pathBufs[j].GetString(i)
+			}
+			v, vNull := json.BinaryJSON{}, valBufs[j].IsNull(i)
+			if !vNull {
+				v = valBufs[j].GetJSON(i)
+			}
+			res, isNull, err = b.appendJSONArray(res, s, sNull, v, vNull)
```
Comment (wshwsh12) [inline expression/builtin_json_vec.go:1032]: If sNull is true, set isNull is true and break the loop.
If vNull is true, set `v = json.CreateBinary(nil)`.
So we can only transfer 3 parameters for function `appendJSONArray`

Comment (qw4990) [reply expression/builtin_json_vec.go]: Fixed. PTAL

**Pull request merged**